src/dolphin/install/installation.ts

```diff
@@ -176,11 +176,16 @@ export class DolphinInstallation {
   }
 
   private async _isOutOfDate(latestVersion: string): Promise<boolean> {
+    const dolphinVersion = await this.getDolphinVersion();
+    return lt(dolphinVersion, latestVersion);
+  }
+
+  public async getDolphinVersion(): Promise<string> {
     const dolphinPath = await this.findDolphinExecutable();
     const dolphinVersionOut = spawnSync(dolphinPath, ["--version"]).stdout.toString();
     const match = dolphinVersionOut.match(semverRegex);
     const dolphinVersion = match?.[0] ?? "";
-    return lt(dolphinVersion, latestVersion);
+    return dolphinVersion;
```

Comment:
Can we wrap this in a try catch block, returning null if there's no match, and also if an error is thrown?

Reply:
Not sure about wrapping in a try catch as this function only gets run if it is confirmed that a dolphin exec exists.

Reply:
We shouldn't know or care where and when `installation.getDolphinVersion()` is or isn't called. If we're exposing a public function to get the dolphin version, I would expect it to return the version if it exists and null if it doesn't, rather than throwing. The caller shouldn't have to catch the error, null should indicate the installation doesn't exist.